### PR TITLE
an EOS interface with NSE

### PR DESCRIPTION
**Pull request by zingale**
solving for T(e) or T(p) in NSE is trickier than our standard interface because Abar and Zbar are
also a function of rho, T, Ye through the NSE table.  This does a self-consistent inversion of the
EOS together with the NSE table.

### CODE REVIEW

**Comment by zingale** [comment]
this seems to work really well now, but needs a bit of cleaning

**Comment by zingale** [comment]
this is ready, but depends on #1407 so that should be reviewed / merged first

File: nse_tabular/nse_eos.H [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    while (not converged && iter < max_iter) {
+
+        // call NSE table to get Abar
+        nse_state.T = T;
+        nse_state.rho = rho;
+        nse_state.Ye = Ye;
+
+        constexpr bool skip_X_fill{true};
+        nse_interp(nse_state, skip_X_fill);
+        Real abar_old = nse_state.abar;
```
Comment (zhichen3) [inline nse_tabular/nse_eos.H:64]: This `abar_old` looks unncessary?

File: unit_test/test_nse_interp/nse_cell.H [hunk 2]
```diff
@@ -167,5 +167,160 @@ void nse_cell_c()
...
+        eos_state.T = temperature;
+        eos_state.rho = density;
+        eos_state.aux[iye] = nse_state.Ye;
+        eos_state.aux[iabar] = nse_state.abar;
+
+        eos(eos_input_rt, eos_state);
+
+        Real e_orig = eos_state.e;
+
+        // now perturn e and find T, then redo NSE to get abar and finally
```
Comment (zhichen3) [inline unit_test/test_nse_interp/nse_cell.H:219]: perturn -> perturb

File: unit_test/test_nse_interp/nse_cell.H [hunk 3]
```diff
@@ -167,5 +167,160 @@ void nse_cell_c()
...
+        eos_state.T = temperature;
+        eos_state.rho = density;
+        eos_state.aux[iye] = nse_state.Ye;
+        eos_state.aux[iabar] = nse_state.abar;
+
+        eos(eos_input_rt, eos_state);
+
+        Real p_orig = eos_state.p;
+
+        // now perturn p and find T, then redo NSE to get abar and finally
```
Comment (zhichen3) [inline unit_test/test_nse_interp/nse_cell.H:285]: perturn -> perturb

File: nse_tabular/nse_eos.H (hunk 1 above)
Comment (zingale) [reply nse_tabular/nse_eos.H:64]: I think it's needed for the EOS call

Comment (zingale) [reply nse_tabular/nse_eos.H:64]: fixed

File: unit_test/test_nse_interp/nse_cell.H (hunk 2 above)
Comment (zingale) [reply unit_test/test_nse_interp/nse_cell.H]: fixed

File: unit_test/test_nse_interp/nse_cell.H (hunk 3 above)
Comment (zingale) [reply unit_test/test_nse_interp/nse_cell.H]: fixed

File: nse_tabular/nse_eos.H (hunk 1 above)
Comment (zhichen3) [reply nse_tabular/nse_eos.H:64]: I mean why can't we just do `nse_state.abar` in line 71

Comment (zingale) [reply nse_tabular/nse_eos.H:64]: oh.  we could.  I didn't think it mattered much.

**Review by zhichen3 — APPROVED** [review]

**Pull request merged**